src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -843,21 +811,17 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
 
             // UNDONE: This is a naive quadratic algorithm; there is a linear algorithm that works. Consider using it.
             var candidates = result.Results;
-            for (int i = 0; i < candidates.Count; ++i)
+            for (int i = 1; i < candidates.Count; ++i)
             {
                 if (candidates[i].Kind != OperatorAnalysisResultKind.Applicable)
                 {
                     continue;
                 }
 
                 // Is this applicable operator better than every other applicable method?
-                for (int j = 0; j < candidates.Count; ++j)
+                for (int j = 0; j < i; ++j)
                 {
-                    if (i == j)
-                    {
-                        continue;
-                    }
-                    if (candidates[j].Kind == OperatorAnalysisResultKind.Inapplicable)
+                    if (candidates[j].Kind != OperatorAnalysisResultKind.Applicable)
                     {
                         continue;
                     }
```

Comment:
I believe it is not transitive, so I'm reverting 824.

---
In reply to: [78060790](https://github.com/dotnet/roslyn/pull/13648#discussion_r78060790) [](ancestors = 78060790,78049650,78047238,78035627)